src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -644,7 +642,7 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
             tmGL.set_ledgerseq(mSeq);
         JLOG(m_journal.trace()) << "Sending header request to "
                                 << (peer ? "selected peer" : "all peers");
-        sendRequest(tmGL, peer);
+        mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
```

Comment:
It's possible to use some metaprogramming to compute the type code from the message type. It makes `sendRequest` a little more convenient and reduces the opportunities for error. This patch compiles against `912474b`.

```diff
diff --git a/src/ripple/app/ledger/impl/InboundLedger.cpp b/src/ripple/app/ledger/impl/InboundLedger.cpp
index 84d2e7c1a..c5d96fde6 100644
--- a/src/ripple/app/ledger/impl/InboundLedger.cpp
+++ b/src/ripple/app/ledger/impl/InboundLedger.cpp
@@ -645 +645 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
-        mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+        mPeerSet->sendRequest(tmGL, peer);
@@ -682 +682 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
-            mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+            mPeerSet->sendRequest(tmGL, peer);
@@ -727,2 +727 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
-                        mPeerSet->sendRequest(
-                            tmGL, protocol::mtGET_LEDGER, peer);
+                        mPeerSet->sendRequest(tmGL, peer);
@@ -755 +754 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
-            mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+            mPeerSet->sendRequest(tmGL, peer);
@@ -792 +791 @@ InboundLedger::trigger(std::shared_ptr<Peer> const& peer, TriggerReason reason)
-                    mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+                    mPeerSet->sendRequest(tmGL, peer);
diff --git a/src/ripple/app/ledger/impl/LedgerDeltaAcquire.cpp b/src/ripple/app/ledger/impl/LedgerDeltaAcquire.cpp
index e722dfaed..3c40feb9c 100644
--- a/src/ripple/app/ledger/impl/LedgerDeltaAcquire.cpp
+++ b/src/ripple/app/ledger/impl/LedgerDeltaAcquire.cpp
@@ -116,2 +116 @@ LedgerDeltaAcquire::trigger(std::size_t limit, ScopedLockType& sl)
-                peerSet_->sendRequest(
-                    request, protocol::mtREPLAY_DELTA_REQ, peer);
+                peerSet_->sendRequest(request, peer);
diff --git a/src/ripple/app/ledger/impl/SkipListAcquire.cpp b/src/ripple/app/ledger/impl/SkipListAcquire.cpp
index c0cdb28a9..9c0b39f46 100644
--- a/src/ripple/app/ledger/impl/SkipListAcquire.cpp
+++ b/src/ripple/app/ledger/impl/SkipListAcquire.cpp
@@ -109,2 +109 @@ SkipListAcquire::trigger(std::size_t limit, ScopedLockType& sl)
-                peerSet_->sendRequest(
-                    request, protocol::mtPROOF_PATH_REQ, peer);
+                peerSet_->sendRequest(request, peer);
diff --git a/src/ripple/app/ledger/impl/TransactionAcquire.cpp b/src/ripple/app/ledger/impl/TransactionAcquire.cpp
index cddd059ce..146bc7f75 100644
--- a/src/ripple/app/ledger/impl/TransactionAcquire.cpp
+++ b/src/ripple/app/ledger/impl/TransactionAcquire.cpp
@@ -26,0 +27 @@
+#include <ripple/overlay/impl/ProtocolMessage.h>
@@ -138 +139 @@ TransactionAcquire::trigger(std::shared_ptr<Peer> const& peer)
-        mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+        mPeerSet->sendRequest(tmGL, peer);
@@ -172 +173 @@ TransactionAcquire::trigger(std::shared_ptr<Peer> const& peer)
-        mPeerSet->sendRequest(tmGL, protocol::mtGET_LEDGER, peer);
+        mPeerSet->sendRequest(tmGL, peer);
diff --git a/src/ripple/overlay/PeerSet.h b/src/ripple/overlay/PeerSet.h
index cce5e6b35..aa30f63bc 100644
--- a/src/ripple/overlay/PeerSet.h
+++ b/src/ripple/overlay/PeerSet.h
@@ -27,0 +28 @@
+#include "overlay/impl/ProtocolMessage.h"
@@ -62,0 +64,7 @@ public:
+    template <typename MessageType>
+    void
+    sendRequest(MessageType const& message, std::shared_ptr<Peer> const& peer)
+    {
+        this->sendRequest(message, protocolMessageType(message), peer);
+    }
+
diff --git a/src/ripple/overlay/impl/ProtocolMessage.h b/src/ripple/overlay/impl/ProtocolMessage.h
index 593b2ffd3..78fc7350a 100644
--- a/src/ripple/overlay/impl/ProtocolMessage.h
+++ b/src/ripple/overlay/impl/ProtocolMessage.h
@@ -33,0 +34 @@
+#include <ripple.pb.h>
@@ -38,0 +40,18 @@ namespace ripple {
+inline protocol::MessageType
+protocolMessageType(protocol::TMGetLedger const&)
+{
+    return protocol::mtGET_LEDGER;
+}
+
+inline protocol::MessageType
+protocolMessageType(protocol::TMReplayDeltaRequest const&)
+{
+    return protocol::mtREPLAY_DELTA_REQ;
+}
+
+inline protocol::MessageType
+protocolMessageType(protocol::TMProofPathRequest const&)
+{
+    return protocol::mtPROOF_PATH_REQ;
+}
+
```

Reply:
Thanks so much! changed.